aries_cloudagent/revocation/models/issuer_revocation_record.py

```diff
@@ -0,0 +1,245 @@
+"""Issuer revocation registry storage handling."""
+
+import json
+import uuid
+from typing import Sequence
+
+import indy.anoncreds
+import indy.blob_storage
+from marshmallow import fields
+
+from ...config.injection_context import InjectionContext
+from ...messaging.models.base_record import BaseRecord, BaseRecordSchema
+from ...ledger.base import BaseLedger
+from ...wallet.base import BaseWallet
+
+from ..error import RevocationError
+
+from .revocation_registry import RevocationRegistry
+
+DEFAULT_REGISTRY_SIZE = 100
+
+
+class IssuerRevocationRecord(BaseRecord):
+    """Class for managing local issuing revocation registries."""
+
+    class Meta:
+        """IssuerRevocationRecord metadata."""
+
+        schema_class = "IssuerRevocationRecordSchema"
+
+    RECORD_ID_NAME = "record_id"
+    RECORD_TYPE = "issuer_revoc"
+    LOG_STATE_FLAG = "debug.revocation"
+    CACHE_ENABLED = False
+    TAG_NAMES = {
+        "cred_def_id",
+        "issuance_type",
+        "issuer_did",
+        "revoc_def_type",
+        "revoc_reg_id",
+        "state",
+    }
+
+    ISSUANCE_BY_DEFAULT = "ISSUANCE_BY_DEFAULT"
+    ISSUANCE_ON_DEMAND = "ISSUANCE_ON_DEMAND"
+
+    REVOC_DEF_TYPE_CL = "CL_ACCUM"
+
+    STATE_INIT = "init"
+    STATE_GENERATED = "generated"
+    STATE_PUBLISHED = "published"
+    STATE_ACTIVE = "active"
+    STATE_FULL = "full"
+
+    def __init__(
+        self,
+        *,
+        record_id: str = None,
+        state: str = None,
+        cred_def_id: str = None,
+        error_msg: str = None,
+        issuance_type: str = None,
+        issuer_did: str = None,
+        max_cred_num: int = None,
+        revoc_def_type: str = None,
+        revoc_reg_id: str = None,
+        revoc_reg_def: dict = None,
+        revoc_reg_entry: dict = None,
+        tag: str = None,
+        tails_hash: str = None,
+        tails_local_path: str = None,
+        tails_public_uri: str = None,
+        **kwargs,
+    ):
+        """Initialize the issuer revocation registry record."""
+        super(IssuerRevocationRecord, self).__init__(
+            record_id, state=state or self.STATE_INIT, **kwargs
+        )
+        self.cred_def_id = cred_def_id
+        self.error_msg = error_msg
+        self.issuance_type = issuance_type or self.ISSUANCE_BY_DEFAULT
```

Comment:
Issuance by default is way faster on revocation because it doesn't need another round trip to the ledger. Suggest making this not optionable.